superset/models/core.py

```diff
@@ -743,6 +744,16 @@ def get_perm(self):
         return (
             "[{obj.database_name}].(id:{obj.id})").format(obj=self)
 
+    def has_table(self, table):
+        engine = self.get_sqla_engine()
+        return engine.dialect.has_table(
+            engine, table.table_name, table.schema or None)
+
+    def get_dialect(self):
+        sqla_url = url.make_url(self.sqlalchemy_uri_decrypted)
+        entrypoint = sqla_url._get_entrypoint()
```

Comment:
why not `return self.get_sqla_engine().dialect` ?

Reply:
There's a bit more overhead to it, plus creating an engine just to get the dialect and then ignoring the instance seems wasteful.

These few lines are what `create_engine` does to determine dialect anyway.

Reply:
I hate calling private functions in external libs though... there's no guarantees as to whether the api may change in minor version changes.

To mitigate the cost we could change the use of `get_sqla_engine` by using a cached `engine` `@property`. I think it's only a matter of adding a decorator.

Reply:
check out the `@memoize` decorator in `superset.utils`

Reply:
also for the sake of moving forward I don't mind if you just call `get_sqla_engine` here and save the `sqla_engine` memoized property for another PR

Reply:
Sounds good. I was looking for a way to cache the `sqla_engine` because that function does get called frequently.

Reply:
Oh, turns out we can just call `sqla_url.get_dialect()`